### PR TITLE
New format for defaults file

### PR DESCRIPTION
**Pull request by sandain**
This removes an avenue of attack mentioned in Issue #129.

### CODE REVIEW

**Comment by zanix** [comment]
Nice, this branch has been working pretty well for me so far. I'll try to test the defaults converter soon.

**Comment by sandain** [comment]
Let me know what you think of the updater code.  I think it is a fairly simple idea that can be expanded in the future without junking up the Makefile, mscs or msctl scripts

**Comment by zanix** [comment]
It's a nice approach and a good idea. Keeps the scripts clean.

File: update.d/10-old-defaults [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    $_ =~ s/[\r\n]//g;
+    if ($_ =~ /(\w+)\s*=\s*(.*)/) {
+      if (defined $mscs{$1}) {
+        print "$mscs{$1}=$2\n";
+      }
+      else {
+        print "\x23 $_\n";
+      }
+    }
+    elsif ($_ =~ /^\s*\x23/) {
```
Comment (sandain) [inline update.d/10-old-defaults]: This should probably be the first condition.

**Comment by sandain** [comment]
I should go back and add comments for: the ascii codes (\x23 is ", \x27 is ', and \x23 is #), and the what the regex are doing in the update script.

File: msctl [hunk 2]
```diff
@@ -391,12 +391,14 @@ getValue() {
   if [ -e "$1" ]; then
     # Find the key/value combo.
     KEY=$($PERL -ne '
-      $_ =~ s/[\x22\x27\r]//;
-      if ($_ =~ /^('$2')=.*$/i) { print lc $1; }
+      $_ =~ s/[\x22\x27\r\n]//g;
+      $_ =~ s/\x23.*//;
+      if ($_ =~ /^('$2')\s*=\s*.*$/i) { print lc $1; }
```
Comment (sandain) [inline msctl]: Allow spaces at beginning of line:
```
if ($_ =~ /^\s*('$2')\s*=\s*.*$/i) { print lc $1; }
```

File: msctl [hunk 3]
```diff
@@ -391,12 +391,14 @@ getValue() {
...
     ' $1)
     VALUE=$($PERL -ne '
-      $_ =~ s/[\x22\x27\r]//;
-      if ($_ =~ /^'$2'=(.*)$/i) { print $1; }
+      $_ =~ s/[\x22\x27\r\n]//g;
+      $_ =~ s/\x23.*//;
+      if ($_ =~ /^'$2'\s*=\s*(.*)$/i) { print $1; }
```
Comment (sandain) [inline msctl]: Same here:
```
if ($_ =~ /^\s*'$2'\s*=\s*(.*)$/i) { print $1; }
```

**Comment by sandain** [comment]
I created a new v4 branch with pull request #134. Closing.

**Pull request closed without merging**